test/bin/scenario.sh

```diff
@@ -72,31 +72,62 @@ copy_file_to_vm() {
 }
 
 sos_report() {
+    local -r junit="${1:-false}"
+
     if "${SKIP_SOS}"; then
         echo "Skipping sos reports"
+        if "${junit}"; then
+            record_junit "post_setup" "sos-report" "SKIP"
+        fi
         return
     fi
+
+    echo "Creating sos reports"
     local vmname
     local ip
-    echo "Creating sos reports"
+    local scenario_result=0
     for vmdir in "${SCENARIO_INFO_DIR}"/"${SCENARIO}"/vms/*; do
         if [ ! -d "${vmdir}" ]; then
             # skip log files, etc.
             continue
         fi
+
         vmname=$(basename "${vmdir}")
         ip=$(get_vm_property "${vmname}" ip)
         if [ -z "${ip}" ]; then
             # skip hosts without NICs
             # FIXME: use virsh to copy sos report files
+            if "${junit}"; then
+                record_junit "${vmname}" "sos-report" "SKIP"
+            fi
             continue
         fi
-        # Some scenarios do not start with MicroShift installed, so we
-        # can't rely on the wrapper being there or working if it
-        # is. Copy the script to the host, just in case, along with a
-        # wrapper that knows how to execute it or the installed
-        # version.
-        cat - >/tmp/sos-wrapper.sh <<EOF
+
+        if ! sos_report_for_vm "${vmdir}" "${vmname}" "${ip}"; then
+            scenario_result=1
+            if "${junit}"; then
+                record_junit "${vmname}" "sos-report" "FAILED"
```

Comment:
I wonder if we really want to fail a job if only the SOS gather fails. Maybe? Let's keep it and see how it goes.

Reply:
Actually that would be useful. For example, if nightlies wouldn't stop after iso-boot because of this ([example - 4.14 metal nightly](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-microshift-release-4.14-ocp-metal-nightly/1722955034682986496/artifacts/ocp-metal-nightly/openshift-microshift-infra-iso-boot/artifacts/scenario-info/el92-src@fdo/boot.log)) then at least we would have some test runs.
However, to ignore them I guess we would have to add two junit entries: fail and success, so prow interprets them as flaky?
And also make sure that at the end of the final trap, we return the status of the overall procedure, not sos-report. On the other hand, when we have some syntax errors, we want it to fail...